_posts/2016-01-22-let-vs-instance.markdown

```diff
@@ -0,0 +1,202 @@
+    expect(user_2.email).to eq('bar@ombulabs.com')
+  end
+
+  it 'combined test that checks all users' do
+    expect(user.email).to eq('foo@ombulabs.com')
+    expect(user_2.username).to eq('barbar')
+  end
+end
+```
+
+So let's run both specs separate and have a look on the output.
+
+```bash
+$ bin/rspec spec/models/user_instance_spec.rb
+...
+
+Finished in 0.087 seconds (files took 2.87 seconds to load)
+3 examples, 0 failures
+```
+
+```bash
+$ bin/rspec spec/models/user_let_spec.rb
+...
+
+Finished in 0.06692 seconds (files took 3.06 seconds to load)
+3 examples, 0 failures
+```
+
+What we already can see is that the `let`-tests are faster.
+
+If we have a look at the test logs, we can see why that is.
+
+Logoutput of `user_instance_spec.rb`:
+
+```bash
+SQL (2.6ms)  INSERT INTO `users` (`email`, `username`, `created_at`, `updated_at`) VALUES ('foo@ombulabs.com', 'foofoo', '2016-01-22 16:44:42', '2016-01-22 16:44:42')
+SQL (0.3ms)  INSERT INTO `users` (`email`, `username`, `created_at`, `updated_at`) VALUES ('bar@ombulabs.com', 'barbar', '2016-01-22 16:44:42', '2016-01-22 16:44:42')
+SQL (0.5ms)  INSERT INTO `users` (`email`, `username`, `created_at`, `updated_at`) VALUES ('foo@ombulabs.com', 'foofoo', '2016-01-22 16:44:42', '2016-01-22 16:44:42')
+SQL (0.4ms)  INSERT INTO `users` (`email`, `username`, `created_at`, `updated_at`) VALUES ('bar@ombulabs.com', 'barbar', '2016-01-22 16:44:42', '2016-01-22 16:44:42')
+SQL (0.3ms)  INSERT INTO `users` (`email`, `username`, `created_at`, `updated_at`) VALUES ('foo@ombulabs.com', 'foofoo', '2016-01-22 16:44:42', '2016-01-22 16:44:42')
+SQL (0.3ms)  INSERT INTO `users` (`email`, `username`, `created_at`, `updated_at`) VALUES ('bar@ombulabs.com', 'barbar', '2016-01-22 16:44:42', '2016-01-22 16:44:42')
+```
+
+As we can see here, that apparently for each of my tests all records are being created.
+This is not the case for `let` examples, if we have a look at the log output.
+
+Logoutput of `user_let_spec.rb`:
+
+```bash
+SQL (0.7ms)  INSERT INTO `users` (`email`, `username`, `created_at`, `updated_at`) VALUES ('foo@ombulabs.com', 'foofoo', '2016-01-22 16:47:45', '2016-01-22 16:47:45')
+SQL (0.4ms)  INSERT INTO `users` (`email`, `username`, `created_at`, `updated_at`) VALUES ('bar@ombulabs.com', 'barbar', '2016-01-22 16:47:45', '2016-01-22 16:47:45')
+SQL (0.3ms)  INSERT INTO `users` (`email`, `username`, `created_at`, `updated_at`) VALUES ('foo@ombulabs.com', 'foofoo', '2016-01-22 16:47:45', '2016-01-22 16:47:45')
+SQL (0.3ms)  INSERT INTO `users` (`email`, `username`, `created_at`, `updated_at`) VALUES ('bar@ombulabs.com', 'barbar', '2016-01-22 16:47:45', '2016-01-22 16:47:45')
+```
+
+As we can see in the log output, we count 6 SQL Inserts for the instance example and just 4 SQL Inserts for the `let` example. This is because the variables configured with `let` will be loaded if we directly them. This behaviour is called lazy loading and forgives you small mistakes when writing your tests.
```

Comment:
I think you may have missed a word in this sentence: " This is because the variables configured with `let` will be loaded if we directly them.". Maybe "use them directly" or "set them directly"? 

Also, "forgives you small mistakes" should probably be "forgives your small mistakes".